### PR TITLE
feat(components): add table component

### PR DESCRIPTION
**Pull request by richbachman**
## Description of the change

Sets up the un-styled composable Table components and parts. Includes stories and tests.

## Testing the change

- [ ] Write your testing instructions here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.

### CODE REVIEW

File: packages/components/src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          <Td>dietr-ehler@arketmay.com</Td>
+          <Td>Closed</Td>
+          <Td>Full relocation</Td>
+        </Tr>
+      </TBody>
+    </Table>
+  );
+};
+
+export const Default = Template.bind({});
```
Comment (RohitRajendran) [inline packages/components/src/components/Table/Table.stories.tsx:33]: Thoughts on adding an example with React Table integrated?

Comment (richbachman) [reply packages/components/src/components/Table/Table.stories.tsx:33]: Yeah. I can drop a story in for that.

Comment (richbachman) [reply packages/components/src/components/Table/Table.stories.tsx:40]: @RohitRajendran example story has been added.

**Review by RohitRajendran — APPROVED** [review]
I think it would be good to add a story description for the React Table story as well but other than that looks good!

File: packages/components/src/components/Table/Table.stories.tsx [hunk 2]
```diff
@@ -37,3 +44,99 @@ Default.parameters = {
       "You can use the various table elements to make up your table without having to worry about the styling.",
```
[the remark this answers is not in the document]
Comment (richbachman) [reply packages/components/src/components/Table/Table.stories.tsx:48]: I also added sorting to the table example to make it more like a datagrid.

[the remark this answers is not in the document]
Comment (blumaa) [reply packages/components/src/components/Table/Table.stories.tsx:48]: What about filters and pagination? :)

**Pull request merged**